vaadin-spring-tests/test-spring-security-fusion/src/test/java/com/vaadin/flow/spring/fusionsecurity/SecurityIT.java

```diff
@@ -36,12 +37,18 @@ public void tearDown() {
     private void checkForBrowserErrors() {
         checkLogsForErrors(msg -> {
             return msg.contains(
-                    "admin-only/secret.txt - Failed to load resource: the server responded with a status of 403");
+                    "admin-only/secret.txt - Failed to load resource: the " +
+                            "server responded with a status of 403") ||
+                    msg.contains("webpack-internal://");
         });
     }
 
-    private void logout() {
-        if (!$(ButtonElement.class).attribute("id", "logout").exists()) {
+    protected void logout() {
+        ElementQuery<TestBenchElement> mainViewQuery = $("*").attribute("id",
+                "main-view");
+        if (!mainViewQuery.exists() ||
+                mainViewQuery.get(0).$(ButtonElement.class)
```

Comment:
why is this condition? I read it as if main doesn't exist or if the logout button exists, then open the mainview?